### PR TITLE
[Debt] Improve RoD seeding some more

### LINKED ISSUE #10165: ♻️ Improve seeding for RoD some more
## ♻️ Debt/Refactor

This is a follow-up to #10110 . It adds a few more details to the candidates that get seeded. Basically, it adds seeding data for:

- Veteran "verification"
- Priority "verification"
- Removing candidates
- Placing candidates

And a few other things that @brindasasi added.

## 🕵️ Details

In this document I've outline 7 job advertisements and 8 candidate "personas". The candidates have applications to all the published jobs.

[ROD Seeding Data.docx](https://github.com/GCTC-NTGC/gc-digital-talent/files/15099736/ROD.Seeding.Data.docx)

The items highlighted in yellow are the things that need to be added. The rest were done in #10110.

Ask @gobyrne if you have any questions.

## ✅ Acceptance Criteria

- [ ] New seeding data seeds
- [ ] Change EX-03 generic job title to "Digital_Leader_EX-03"

```[tasklist]
### Blockers
- [ ] #10110
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10059
- [ ] #9198
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/9140
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10417
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10418
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10419
```

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #10165

## 👋 Introduction

Updates the seeder more per instruction. Also updates the key in enums as well as runs a migration to keep the prod values in sync. 

## 🧪 Testing

1. Run `make seed-fresh`
2. Check out some candidates
3. Observe the migration

### CODE REVIEW

File: api/database/seeders/PoolCandidateTestSeeder.php [hunk 1]
```diff
@@ -57,7 +67,12 @@ public function run()
             ->asApplicant()
             ->withSkillsAndExperiences()
             ->afterCreating(function (User $user) {
-                $this->applyToAllPools($user, PoolCandidateStatus::QUALIFIED_AVAILABLE, now()->addYears(2));
+                $this->applyToAllPools(
+                    user: $user,
+                    status: PoolCandidateStatus::QUALIFIED_AVAILABLE->name,
+                    expiryDate: now()->addYears(2),
+                    veteranStatus: ClaimVerificationResult::ACCEPTED->name,
```
Comment (brindasasi) [inline api/database/seeders/PoolCandidateTestSeeder.php:74]: This should be unverified as per the document

**Review by brindasasi — CHANGES_REQUESTED** [review]
minor change needed .. other than that looks good!

File: api/database/seeders/PoolCandidateTestSeeder.php (hunk 1 above)
Comment (vd1992) [reply api/database/seeders/PoolCandidateTestSeeder.php]: Correct! https://github.com/GCTC-NTGC/gc-digital-talent/pull/11617/commits/f40bf31b04277e0d5460bb5092b65b5a8e405b03

**Review by brindasasi — APPROVED** [review]
LGTM!

**Pull request merged**